src/main/java/org/sciborgs1155/robot/drive/Drive.java

```diff
@@ -1,263 +1,181 @@
-   *
-   * <p>This method uses {@link Pose2d#log(Pose2d)} to reduce skew.
-   *
-   * @param speeds The desired chassis speeds.
-   */
-  public void drive(ChassisSpeeds speeds) {
-    var target =
-        new Pose2d(
-            speeds.vxMetersPerSecond * Constants.PERIOD,
-            speeds.vyMetersPerSecond * Constants.PERIOD,
-            Rotation2d.fromRadians(speeds.omegaRadiansPerSecond * Constants.PERIOD));
-
-    var twist = new Pose2d().log(target);
-
-    speeds =
-        new ChassisSpeeds(
-            twist.dx / Constants.PERIOD,
-            twist.dy / Constants.PERIOD,
-            twist.dtheta / Constants.PERIOD);
-
-    setModuleStates(kinematics.toSwerveModuleStates(speeds));
-  }
-
-  /**
-   * Sets the swerve ModuleStates.
-   *
-   * @param desiredStates The desired ModuleIO states.
-   */
-  public void setModuleStates(SwerveModuleState[] desiredStates) {
-    if (desiredStates.length != modules.size()) {
-      throw new IllegalArgumentException("desiredStates must have the same length as modules");
-    }
-
-    SwerveDriveKinematics.desaturateWheelSpeeds(desiredStates, MAX_SPEED);
+    setVoltage(rFF+rFB, lFF+lFB);
 
-    for (int i = 0; i < modules.size(); i++) {
-      modules.get(i).setDesiredState(desiredStates[i]);
-    }
   }
 
-  /** Resets the drive encoders to currently read a position of 0. */
-  public void resetEncoders() {
-    modules.forEach(ModuleIO::resetEncoders);
+  public double[] getVelocity(){
+    double[] velocities = {rightEncoder.getVelocity(), leftEncoder.getVelocity()};
+    return velocities;
   }
 
-  /** Zeroes the heading of the robot. */
-  public CommandBase zeroHeading() {
-    return runOnce(imu::reset);
+  public Rotation2d getRotation() {
+    return gyro.getRotation2d();
   }
 
-  /** Returns the pitch of the drive gyro */
-  public double getPitch() {
-    return imu.getPitch();
+  public Pose2d getPose() {
+    return odometry.getEstimatedPosition();
   }
 
-  private SwerveModuleState[] getModuleStates() {
-    return modules.stream().map(ModuleIO::getState).toArray(SwerveModuleState[]::new);
-  }
+  // public void resetPose(){
+    
+  // }
+
+  //may or may not need the abovee method ^^
 
-  private SwerveModulePosition[] getModulePositions() {
-    return modules.stream().map(ModuleIO::getPosition).toArray(SwerveModulePosition[]::new);
+  public double getPoseDegrees() {
+    return getPose().getRotation().getDegrees();
   }
+  
+  // either "right" "Right" "left" or "Left" is available for String side, otherwise returns null
 
-  /** Updates pose estimation based on provided {@link EstimatedRobotPose} */
-  public void updateEstimates(EstimatedRobotPose... poses) {
-    for (int i = 0; i < poses.length; i++) {
-      odometry.addVisionMeasurement(poses[i].estimatedPose.toPose2d(), poses[i].timestampSeconds);
-      field2d.getObject("Cam-" + i + " Est Pose").setPose(poses[i].estimatedPose.toPose2d());
-    }
+  public RelativeEncoder getEncoder(String side){
```

Comment:
WOAH THERE
1. A public method to return internal hardware is almost always a bad idea
2. Using strings as an input and then checking them with if statements is almost always a bad idea
3. Returning null from a function that doesn't need to (as opposed to `Optional` or something similar is almost always a bad idea